test/test_response.py

```diff
@@ -198,7 +200,9 @@ def test_io(self):
         self.assertRaises(IOError, resp3.fileno)
 
     def test_io_closed_consistently(self):
-        hlr = httplib.HTTPResponse(socket.socket())
+        sock = socket.socket()
+        self.addCleanup(sock.close)
```

Comment:
Ah, so the urllib3 test suite uses pytest==3.1.0, which shows the warning. The latest version of pytest==3.1.2 doesn’t show the warning, but running under unittest drops the warning on both occasions. 🤔